schulze_condorcet/schulze_condorcet.py

```diff
@@ -139,13 +184,24 @@ def _subindex(alist: Collection[Container[str]], element: str) -> int:
         winners = _schulze_winners(d, remaining)
         result.append(winners)
 
-    # Return the aggregated preference list in the same format as the input votes are.
-    condensed = Vote(">".join("=".join(level) for level in result))
-    detailed = []
-    for preferred_candidates, rejected_candidates in zip(result, result[1:]):
+    # At last, construct a vote string reflecting the overall preference
+    result_tuple = tuple(tuple(candidate for candidate in level) for level in result)
+    return _recombine_vote(result_tuple)
+
+
+def detailed_result(result: Vote, counts: PairwisePreference) -> List[DetailedResultLevel]:
```

Comment:
I would name this `schulze_evaluate_detailed` or something similar to avoid introducing a completely new name.
Also I would make they parameters the same as `schulze_evaluate`, which means some duplicate computations, but that should not be a big issue (actually returning both things in `schulze_evaluate` probably was on the side of premature optimization).

Reply:
Hmm, but this would mean to mostly copy+paste the current schulze_evaluate function...

Reply:
I don't really understand what is duplicated there, but it should be extractable into a helper function.

Reply:
I have refactored the whole routine in a helper function, which returns all intermediate calculations, so they can be used during construction of the return of the exposed functions.